tree-sitter-stack-graphs/src/util.rs

```diff
@@ -33,3 +40,61 @@ impl std::fmt::Display for DisplayParseErrorsPretty<'_> {
         Ok(())
     }
 }
+
+pub struct TreeSitterCancellationFlag<'a> {
+    flag: Arc<AtomicUsize>,
+    _phantom: PhantomData<&'a ()>,
+}
+
+impl<'a> TreeSitterCancellationFlag<'a> {
+    fn from(cancellation_flag: &'a dyn CancellationFlag) -> Self {
+        let flag = Arc::new(AtomicUsize::new(0));
+        let thread_flag = Arc::downgrade(&flag);
+        // The 'a lifetime on cancellation_flag is changed to a 'static lifetime
+        // so that we can pass it to the polling thread. This is possible, because:
+        //   (1) The lifetime parameter on `TreeSitterCancellationFlag` ensures it does
+        //       not outlive the original reference.
+        //   (1) The thread captures a weak reference to the flag, which ensures that
+        //       `cancellation_flag` are only accessed as long as the flag exists.
+        //   (2) The field `self.flag` is the only other reference to the flag, ensuring
+        //       the flag does not outlive the struct.
+        //   (3) The lifetime parameter `'a` ensures that the struct does not outlive the
+        //       `cancellation_flag` reference.
```

Comment:
Small nit: does the duplicate `(1)` matter, or does this get formatted into a doc comment?

Reply:
Good catch! This is just a regular code comment, so I'll fix that.